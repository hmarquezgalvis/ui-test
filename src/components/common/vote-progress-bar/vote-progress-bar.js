import React, { useEffect, useState } from 'react';
import styled from 'styled-components';

import { ThumbDownIcon, ThumbUpIcon } from '../icons/thumbs';
import { getPercentage } from '../../../helpers/getPercentage';

export const VoteProgressBar = ({ positive = 0, negative = 0}) => {
  const initial = { positive, negative };
  const [{ positive: thumbUp, negative: thumbDown }, setPercents] = useState(initial);

  useEffect(() => {
    const total = positive + negative;
    if (total !== 0) {
      setPercents({
        positive: getPercentage(positive, total, 2),
        negative: getPercentage(negative, total, 2),
      });
    }
  }, [positive, negative]);

  return (
    <VoteProgressBarContainer middle={thumbUp}>
      <ProgressBarItem>
        <ThumbUpIcon altText='Thumbs Up' />
        <Label>{thumbUp}%</Label>
      </ProgressBarItem>
      <ProgressBarItem>
        <Label>{thumbDown}%</Label>
        <ThumbDownIcon altText='Thumbs Down' />
      </ProgressBarItem>
    </VoteProgressBarContainer>
  );
};

const VoteProgressBarContainer = styled.div`
  display: flex;
  width: 100%;
  color: #FFFFFF;
  font-size: 18px;
  line-height: 22px;
  background: ${props => `linear-gradient(90deg, 
    rgba(60, 187, 180, 0.6) 0%, 
    rgba(60, 187, 180, 0.6) ${props.middle}%, 
    rgba(249, 173, 29, 0.6) ${props.middle}%, 
    rgba(249, 173, 29, 0.6) 100%)`};
`;
const ProgressBarItem = styled.div`
  flex: 1;
  display: flex;
  align-items: center;
  padding: 7px 12px;
  &:last-child {
    justify-content: end;
  }
`;
const Label = styled.span`
  margin: 0 5px;
`;
