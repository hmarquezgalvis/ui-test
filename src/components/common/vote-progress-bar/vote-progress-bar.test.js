import { render, screen } from '@testing-library/react';
import { VoteProgressBar } from "./vote-progress-bar"

describe('tests of <VoteProgressBar />', () => {

  it('should render and save the snapshot correctly', () => {
    const { asFragment } = render(
      <VoteProgressBar positive={70} negative={30} />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText(/70.00%/i)).toBeInTheDocument();
    expect(screen.getByText(/30.00%/i)).toBeInTheDocument();
  });

});