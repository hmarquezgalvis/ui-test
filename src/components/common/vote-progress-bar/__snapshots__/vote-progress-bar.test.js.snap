// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests of <VoteProgressBar /> should render and save the snapshot correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ ckZYpz"
  >
    <div
      class="sc-gsnTZi kIJNTc"
    >
      <img
        alt="Thumbs Up"
        src="thumbs-up.svg"
      />
      <span
        class="sc-dkzDqf lbOcmF"
      >
        70.00%
      </span>
    </div>
    <div
      class="sc-gsnTZi kIJNTc"
    >
      <span
        class="sc-dkzDqf lbOcmF"
      >
        30.00%
      </span>
      <img
        alt="Thumbs Down"
        src="thumbs-down.svg"
      />
    </div>
  </div>
</DocumentFragment>
`;
